ssz/tools/io.py

```diff
@@ -0,0 +1,150 @@
+from collections.abc import (
+    Mapping,
+    Sequence,
+)
+
+from eth_utils import (
+    decode_hex,
+    encode_hex,
+)
+
+from ssz.sedes import (
+    BaseSedes,
+    Boolean,
+    ByteList,
+    ByteVector,
+    Container,
+    List,
+    UInt,
+    Vector,
+)
+from ssz.sedes.serializable import (
+    BaseSerializable,
+    MetaSerializable,
+)
+
+
+class FormattedDictIO:
```

Comment:
I would suggest getting rid of this class entirely as it doesn't appear to be needed, in favor of two functions:

> NOTE: this somewhat hinges on the concept of defining an interface for the dumping/loading of basic values

```python
def dump_value(basic_value_handler, value, sedes):
    ...
def parse_value(basic_value_handler, value, sedes):
    ...
```

These functions can then just recursively walk the individual objects.